go/tasks/plugins/hive/execution_state.go

```diff
@@ -153,6 +153,27 @@ func composeResourceNamespaceWithClusterPrimaryLabel(ctx context.Context, tCtx c
 	return core.ResourceNamespace(clusterPrimaryLabel), nil
 }
 
+func createResourceConstraintsSpec(ctx context.Context, _ core.TaskExecutionContext, targetClusterPrimaryLabel core.ResourceNamespace) core.ResourceConstraintsSpec {
```

Comment:
i guess there is a benefit to doing this (provide the spec for every call), the benefit is it could allow dynamic adjustment of quota. One problem is that the resolution may not be deterministic if we allow dynamic adjustment

Reply:
Yes, the addition of this is to enable dynamically adjustable constraints.

Reply:
Could you elaborate a bit on what you mean by "resolution not being deterministic?"  @kumare3

Reply:
we discussed offline. So we are going to call out the caveats and explain the semantics in the documentation